::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2017 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Guillermo Herrero <guillermo.herrero@enea.com>
Date: Wed, 13 Sep 2017 14:15:07 +0200
Subject: [PATCH] Pod Descriptor File compatible reclass structure

Implementation for baremetal-mcp-ocata-ovs-ha scenario

Change-Id: Id6ab5697f993ac9faa019c3c10ba4ed4b7b6db01
Signed-off-by: Guillermo Herrero <guillermo.herrero@enea.com>
---
 .../cluster/all-mcp-ocata-common/opnfv/init.yml    |   1 +
 .../opnfv/pdf/generate_config.py                   |  52 ++++++
 .../all-mcp-ocata-common/opnfv/pdf/pod5.yaml       | 183 +++++++++++++++++++++
 .../opnfv/pdf/pod_config.yml.j2                    |  93 +++++++++++
 .../all-mcp-ocata-common/opnfv/pod_config.yml      |  92 +++++++++++
 .../opnfv/runtime.yml.template                     |  10 +-
 .../baremetal-mcp-ocata-ovs-ha/infra/init.yml      |  22 +--
 .../baremetal-mcp-ocata-ovs-ha/infra/kvm.yml       |  11 +-
 .../baremetal-mcp-ocata-ovs-ha/infra/maas.yml      |  72 ++++----
 .../openstack/compute.yml                          |  12 +-
 .../baremetal-mcp-ocata-ovs-ha/openstack/init.yml  |  72 ++++----
 11 files changed, 520 insertions(+), 100 deletions(-)
 create mode 100755 mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/generate_config.py
 create mode 100644 mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod5.yaml
 create mode 100644 mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod_config.yml.j2
 create mode 100644 mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml

diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
index 28619ba..4cb4f2f 100644
--- a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/init.yml
@@ -1,2 +1,3 @@
 classes:
 - cluster.all-mcp-ocata-common.opnfv.runtime
+- cluster.all-mcp-ocata-common.opnfv.pod_config
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/generate_config.py b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/generate_config.py
new file mode 100755
index 0000000..03014a9
--- /dev/null
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/generate_config.py
@@ -0,0 +1,52 @@
+#!/usr/bin/python
+import argparse
+import ipaddress
+import yaml
+from jinja2 import Environment, FileSystemLoader
+
+PARSER = argparse.ArgumentParser()
+PARSER.add_argument("--yaml", "-y", type=str, required=True)
+PARSER.add_argument("--jinja2", "-j", type=str, required=True)
+ARGS = PARSER.parse_args()
+
+# Processor architecture vs DPKG architecture mapping
+DPKG_ARCH_TABLE = {
+    'aarch64': 'arm64',
+    'x86_64': 'amd64',
+}
+ARCH_DPKG_TABLE = dict(zip(DPKG_ARCH_TABLE.values(), DPKG_ARCH_TABLE.keys()))
+
+# Custom filter to allow simple IP address operations returning
+# a new address from an upper or lower (negative) index
+def ipaddr_index(base_address, index):
+    """Return IP address in given network at given index"""
+    try:
+        base_address_str = unicode(base_address)
+    #pylint: disable=unused-variable
+    except NameError as ex:
+        base_address_str = str(base_address)
+    return ipaddress.ip_address(base_address_str) + int(index)
+
+# Custom filter to convert between processor architecture
+# (as reported by $(uname -m)) and DPKG-style architecture
+def dpkg_arch(arch, to_dpkg=True):
+    """Return DPKG-compatible from processor arch and vice-versa"""
+    if to_dpkg:
+        return DPKG_ARCH_TABLE[arch]
+    else:
+        return ARCH_DPKG_TABLE[arch]
+
+ENV = Environment(loader=FileSystemLoader('./'))
+ENV.filters['ipaddr_index'] = ipaddr_index
+ENV.filters['dpkg_arch'] = dpkg_arch
+
+with open(ARGS.yaml) as _:
+    DICT = yaml.safe_load(_)
+
+# Print dictionary generated from yaml (uncomment for debug)
+#print DICT
+
+# Render template and print generated conf to console
+TEMPLATE = ENV.get_template(ARGS.jinja2)
+#pylint: disable=superfluous-parens
+print(TEMPLATE.render(conf=DICT))
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod5.yaml b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod5.yaml
new file mode 100644
index 0000000..35a9b3e
--- /dev/null
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod5.yaml
@@ -0,0 +1,183 @@
+---
+### Enea ARM POD 5 descriptor file ###
+
+details:
+  pod owner: ENEA AB
+  contact: dan.lilliehorn@enea.com
+  lab: ENEA lab
+  location: Stockholm, Sweden
+  type: production
+  link: https://wiki.opnfv.org/display/pharos/Enea+Hosting
+######################################################
+net_config:
+  oob:
+    interface: 0
+    ip-range: 10.0.8.3-10.0.8.10
+    vlan: 0
+  admin:
+    interface: 1
+    vlan: 0
+    network: 192.168.11.0
+    mask: 24
+  mgmt:
+    interface: 2
+    vlan: 2183
+    network: 172.16.10.0
+    mask: 24
+  storage:
+    interface: 2
+    vlan: 2184
+    network: 10.2.0.0
+    mask: 24
+  private:
+    interface: 2
+    vlan: 2185
+    network: 10.1.0.0
+    mask: 24
+  public:
+    interface: 2
+    vlan: 0
+    network: 10.0.8.0
+    mask: 24
+    gateway: 10.0.8.254
+    dns:
+      - 8.8.8.8
+      - 8.8.4.4
+######################################################
+jumphost:
+  name: jenkins-slave-arm-pod5
+  node:
+    type: baremetal
+    vendor: Cavium
+    model: ThunderX
+    arch: aarch64
+    cpus: 1
+    cpu_cflags: fp asimd evtstrm aes pmull sha1 sha2 crc32
+    cores: 48
+    memory: 128G
+  disks:
+    - name: disk1
+      disk_capacity: 500G
+      disk_type: hdd
+      disk_interface: sata
+      disk_rotation: 7200
+  os: ubuntu-16.04
+  remote_params: &remote_params
+    type: ipmi
+    versions:
+      - 2.0
+    user: admin
+    pass: admin
+  remote_management:
+    <<: *remote_params
+    address: 10.0.100.164
+    mac_address: "00:0F:B7:11:18:82"
+  interfaces:
+    - "00:0F:B7:11:18:82"
+    - "00:0f:b7:04:00:c0"
+    - "00:0f:b7:04:00:c1"
+    - "00:0f:b7:04:00:c2"
+    - "68:05:ca:34:60:e3"
+  fixed_ips:
+    admin: 192.168.11.1
+    mgmt: 172.16.10.1
+    public: 10.0.8.9
+######################################################
+nodes:
+  - name: softiron-3
+    node: &nodeparams
+        type: baremetal
+        vendor: Softiron
+        model: Overdrive 3000
+        arch: aarch64
+        cpus: 1
+        cpu_cflags: fp asimd evtstrm aes pmull sha1 sha2 crc32
+        cores: 8
+        memory: 32G
+    disks: &disks
+      - name: 'disk1'
+        disk_capacity: 128G
+        disk_type: ssd
+        disk_interface: sata
+        disk_rotation: ''
+      - name: 'disk2'
+        disk_capacity: 2048G
+        disk_type: hdd
+        disk_interface: sata
+        disk_rotation: 7200
+    remote_management:
+        <<: *remote_params
+        address: 10.0.8.3
+        mac_address: "e0:ff:f7:00:04:99"
+    interfaces:
+        - "e0:ff:f7:00:04:99"
+        - "e0:ff:f7:00:08:ae"
+        - "e0:ff:f7:00:08:af"
+    fixed_ips:
+      admin: 192.168.11.2
+      mgmt: 172.16.10.141
+      public: 10.0.8.141
+######################################################
+  - name: softiron-4
+    node: *nodeparams
+    disk: *disks
+    remote_management:
+      <<: *remote_params
+      address: 10.0.8.4
+      mac_address: "e0:ff:f7:00:05:00"
+    interfaces:
+      - "e0:ff:f7:00:05:00"
+      - "e0:ff:f7:00:08:b1"
+      - "e0:ff:f7:00:08:b2"
+    fixed_ips:
+      admin: 192.168.11.3
+      mgmt: 172.16.10.142
+      public: 10.0.8.142
+######################################################
+  - name: softiron-5
+    node: *nodeparams
+    disk: *disks
+    remote_management:
+      <<: *remote_params
+      address: 10.0.8.5
+      mac_address: "e0:ff:f7:00:04:a9"
+    interfaces:
+      - "e0:ff:f7:00:04:a9"
+      - "e0:ff:f7:00:08:b1"
+      - "e0:ff:f7:00:08:b2"
+    fixed_ips:
+      admin: 192.168.11.4
+      mgmt: 172.16.10.143
+      public: 10.0.8.143
+######################################################
+  - name: softiron-6
+    node: *nodeparams
+    disk: *disks
+    remote_management:
+      <<: *remote_params
+      address: 10.0.8.6
+      mac_address: "e0:ff:f7:00:04:78"
+    interfaces:
+      - "e0:ff:f7:00:04:a9"
+      - "e0:ff:f7:00:08:c6"
+      - "e0:ff:f7:00:08:c7"
+    fixed_ips:
+      admin: 192.168.11.5
+      mgmt: 172.16.10.101
+      public: 10.0.8.101
+######################################################
+  - name: softiron-7
+    node: *nodeparams
+    disk: *disks
+    remote_management:
+      <<: *remote_params
+      address: 10.0.8.7
+      mac_address: "e0:ff:f7:00:05:19"
+    interfaces:
+      - "e0:ff:f7:00:05:19"
+      - "e0:ff:f7:00:08:cf"
+      - "e0:ff:f7:00:08:d0"
+    fixed_ips:
+      admin: 192.168.11.6
+      mgmt: 172.16.10.102
+      public: 10.0.8.102
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod_config.yml.j2 b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod_config.yml.j2
new file mode 100644
index 0000000..28eba22
--- /dev/null
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pdf/pod_config.yml.j2
@@ -0,0 +1,93 @@
+parameters:
+  _param:
+    # infra service addresses
+    opnfv_infra_config_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('100') }}
+    opnfv_infra_maas_node01_address:  {{ conf['net_config']['mgmt']['network'] | ipaddr_index('2') }}
+    opnfv_infra_maas_node01_deploy_address: {{ conf['net_config']['admin']['network'] | ipaddr_index('2') }}
+    opnfv_infra_maas_node01_external_address: {{ conf['net_config']['public']['network'] | ipaddr_index('2') }}
+    opnfv_infra_compute_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('141') }}
+    opnfv_infra_compute_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('142') }}
+    opnfv_infra_compute_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('143') }}
+    opnfv_infra_kvm_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('140') }}
+    opnfv_infra_kvm_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('141') }}
+    opnfv_infra_kvm_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('142') }}
+    opnfv_infra_kvm_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('143') }}
+
+    opnfv_openstack_gateway_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('124') }}
+    opnfv_openstack_gateway_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('125') }}
+    opnfv_openstack_gateway_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('126') }}
+    opnfv_openstack_gateway_node01_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('6') }}
+    opnfv_openstack_gateway_node02_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('7') }}
+    opnfv_openstack_gateway_node03_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('9') }}
+    opnfv_openstack_proxy_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('80') }}
+    opnfv_openstack_proxy_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('81') }}
+    opnfv_openstack_proxy_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('82') }}
+    opnfv_openstack_control_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('10') }}
+    opnfv_openstack_control_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('11') }}
+    opnfv_openstack_control_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('12') }}
+    opnfv_openstack_control_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('13') }}
+    opnfv_openstack_database_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('50') }}
+    opnfv_openstack_database_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('51') }}
+    opnfv_openstack_database_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('52') }}
+    opnfv_openstack_database_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('53') }}
+    opnfv_openstack_message_queue_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('40') }}
+    opnfv_openstack_message_queue_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('41') }}
+    opnfv_openstack_message_queue_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('42') }}
+    opnfv_openstack_message_queue_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('43') }}
+    opnfv_openstack_telemetry_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('75') }}
+    opnfv_openstack_telemetry_node01_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('76') }}
+    opnfv_openstack_telemetry_node02_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('77') }}
+    opnfv_openstack_telemetry_node03_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('78') }}
+    opnfv_openstack_compute_node01_single_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('101') }}
+    opnfv_openstack_compute_node02_single_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('102') }}
+    opnfv_openstack_compute_node03_single_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('103') }}
+    opnfv_openstack_compute_node01_control_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('101') }}
+    opnfv_openstack_compute_node02_control_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('102') }}
+    opnfv_openstack_compute_node03_control_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('103') }}
+    opnfv_openstack_compute_node01_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('101') }}
+    opnfv_openstack_compute_node02_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('102') }}
+    opnfv_openstack_compute_node03_tenant_address: {{ conf['net_config']['private']['network'] | ipaddr_index('103') }}
+    opnfv_openstack_compute_node01_external_address: {{ conf['net_config']['public']['network'] | ipaddr_index('101') }}
+    opnfv_openstack_compute_node02_external_address: {{ conf['net_config']['public']['network'] | ipaddr_index('102') }}
+
+    # opendaylight options
+    #opnfv_opendaylight_server_node01_single_address: {{ conf['net_config']['mgmt']['network'] | ipaddr_index('111') }}
+
+    opnfv_name_servers: {{ conf['net_config']['public']['dns'] | default('8.8.4.4', true) }}
+    opnfv_net_mgmt_vlan: {{ conf['net_config']['mgmt']['vlan'] }}
+    opnfv_net_tenant_vlan: {{ conf['net_config']['private']['vlan'] }}
+
+    opnfv_maas_node01_architecture: '{{ conf['nodes'][0]['node']['arch'] | dpkg_arch }}/generic'
+    opnfv_maas_node01_power_address: {{ conf['nodes'][0]['remote_management']['address'] }}
+    opnfv_maas_node01_power_type: {{ conf['nodes'][0]['remote_management']['type'] }}
+    opnfv_maas_node01_power_user: {{ conf['nodes'][0]['remote_management']['user'] }}
+    opnfv_maas_node01_power_password: {{ conf['nodes'][0]['remote_management']['pass'] }}
+    opnfv_maas_node01_interface_mac: '{{ conf['nodes'][0]['interfaces'][1] }}'
+
+    opnfv_maas_node02_architecture: '{{ conf['nodes'][1]['node']['arch'] | dpkg_arch }}/generic'
+    opnfv_maas_node02_power_address: {{ conf['nodes'][1]['remote_management']['address'] }}
+    opnfv_maas_node02_power_type: {{ conf['nodes'][1]['remote_management']['type'] }}
+    opnfv_maas_node02_power_user: {{ conf['nodes'][1]['remote_management']['user'] }}
+    opnfv_maas_node02_power_password: {{ conf['nodes'][1]['remote_management']['pass'] }}
+    opnfv_maas_node02_interface_mac: '{{ conf['nodes'][1]['interfaces'][1] }}'
+
+    opnfv_maas_node03_architecture: '{{ conf['nodes'][2]['node']['arch'] | dpkg_arch }}/generic'
+    opnfv_maas_node03_power_address: {{ conf['nodes'][2]['remote_management']['address'] }}
+    opnfv_maas_node03_power_type: {{ conf['nodes'][2]['remote_management']['type'] }}
+    opnfv_maas_node03_power_user: {{ conf['nodes'][2]['remote_management']['user'] }}
+    opnfv_maas_node03_power_password: {{ conf['nodes'][2]['remote_management']['pass'] }}
+    opnfv_maas_node03_interface_mac: '{{ conf['nodes'][2]['interfaces'][1] }}'
+
+    opnfv_maas_node04_architecture: '{{ conf['nodes'][3]['node']['arch'] | dpkg_arch }}/generic'
+    opnfv_maas_node04_power_address: {{ conf['nodes'][3]['remote_management']['address'] }}
+    opnfv_maas_node04_power_type: {{ conf['nodes'][3]['remote_management']['type'] }}
+    opnfv_maas_node04_power_user: {{ conf['nodes'][3]['remote_management']['user'] }}
+    opnfv_maas_node04_power_password: {{ conf['nodes'][3]['remote_management']['pass'] }}
+    opnfv_maas_node04_interface_mac: '{{ conf['nodes'][3]['interfaces'][1] }}'
+
+    opnfv_maas_node05_architecture: '{{ conf['nodes'][4]['node']['arch'] | dpkg_arch }}/generic'
+    opnfv_maas_node05_power_address: {{ conf['nodes'][4]['remote_management']['address'] }}
+    opnfv_maas_node05_power_type: {{ conf['nodes'][4]['remote_management']['type'] }}
+    opnfv_maas_node05_power_user: {{ conf['nodes'][4]['remote_management']['user'] }}
+    opnfv_maas_node05_power_password: {{ conf['nodes'][4]['remote_management']['pass'] }}
+    opnfv_maas_node05_interface_mac: '{{ conf['nodes'][4]['interfaces'][1] }}'
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml
new file mode 100644
index 0000000..1030595
--- /dev/null
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/pod_config.yml
@@ -0,0 +1,92 @@
+parameters:
+  _param:
+    # infra service addresses
+    opnfv_infra_config_address: 10.167.4.100
+    opnfv_infra_config_deploy_address: 10.20.0.2
+    opnfv_infra_maas_node01_address: 10.167.4.3
+    opnfv_infra_maas_node01_deploy_address: 192.168.11.3
+    opnfv_infra_maas_node01_external_address: 10.16.0.3
+    opnfv_infra_compute_node01_address: 10.167.4.141
+    opnfv_infra_compute_node02_address: 10.167.4.142
+    opnfv_infra_compute_node03_address: 10.167.4.143
+    opnfv_infra_kvm_address: 10.167.4.140
+    opnfv_infra_kvm_node01_address: 10.167.4.141
+    opnfv_infra_kvm_node02_address: 10.167.4.142
+    opnfv_infra_kvm_node03_address: 10.167.4.143
+
+    opnfv_openstack_gateway_node01_address: 10.167.4.124
+    opnfv_openstack_gateway_node02_address: 10.167.4.125
+    opnfv_openstack_gateway_node03_address: 10.167.4.126
+    opnfv_openstack_gateway_node01_tenant_address: 10.1.0.6
+    opnfv_openstack_gateway_node02_tenant_address: 10.1.0.7
+    opnfv_openstack_gateway_node03_tenant_address: 10.1.0.9
+    opnfv_openstack_proxy_address: 10.167.4.80
+    opnfv_openstack_proxy_node01_address: 10.167.4.81
+    opnfv_openstack_proxy_node02_address: 10.167.4.82
+    opnfv_openstack_control_address: 10.167.4.10
+    opnfv_openstack_control_node01_address: 10.167.4.11
+    opnfv_openstack_control_node02_address: 10.167.4.12
+    opnfv_openstack_control_node03_address: 10.167.4.13
+    opnfv_openstack_database_address: 10.167.4.50
+    opnfv_openstack_database_node01_address: 10.167.4.51
+    opnfv_openstack_database_node02_address: 10.167.4.52
+    opnfv_openstack_database_node03_address: 10.167.4.53
+    opnfv_openstack_message_queue_address: 10.167.4.40
+    opnfv_openstack_message_queue_node01_address: 10.167.4.41
+    opnfv_openstack_message_queue_node02_address: 10.167.4.42
+    opnfv_openstack_message_queue_node03_address: 10.167.4.43
+    opnfv_openstack_telemetry_address: 10.167.4.75
+    opnfv_openstack_telemetry_node01_address: 10.167.4.76
+    opnfv_openstack_telemetry_node02_address: 10.167.4.77
+    opnfv_openstack_telemetry_node03_address: 10.167.4.78
+    opnfv_openstack_compute_node01_single_address: 10.167.4.101
+    opnfv_openstack_compute_node02_single_address: 10.167.4.102
+    opnfv_openstack_compute_node03_single_address: 10.167.4.103
+    opnfv_openstack_compute_node01_control_address: 10.167.4.101
+    opnfv_openstack_compute_node02_control_address: 10.167.4.102
+    opnfv_openstack_compute_node03_control_address: 10.167.4.103
+    opnfv_openstack_compute_node01_tenant_address: 10.1.0.101
+    opnfv_openstack_compute_node02_tenant_address: 10.1.0.102
+    opnfv_openstack_compute_node03_tenant_address: 10.1.0.103
+    opnfv_openstack_compute_node01_external_address: 172.30.10.2
+    opnfv_openstack_compute_node02_external_address: 172.30.10.3
+
+
+    opnfv_name_servers: ['8.8.8.8', '8.8.4.4']
+    opnfv_net_mgmt_vlan: 300
+    opnfv_net_tenant_vlan: 302
+
+    opnfv_maas_node01_architecture: amd64/generic
+    opnfv_maas_node01_power_address: 172.30.8.75
+    opnfv_maas_node01_power_type: ipmi
+    opnfv_maas_node01_power_user: admin
+    opnfv_maas_node01_power_password: octopus
+    opnfv_maas_node01_interface_mac: "00:25:b5:a0:00:2a"
+
+    opnfv_maas_node02_architecture: amd64/generic
+    opnfv_maas_node02_power_address: 172.30.8.65
+    opnfv_maas_node02_power_type: ipmi
+    opnfv_maas_node02_power_user: admin
+    opnfv_maas_node02_power_password: octopus
+    opnfv_maas_node02_interface_mac: "00:25:b5:a0:00:3a"
+
+    opnfv_maas_node03_architecture: amd64/generic
+    opnfv_maas_node03_power_address: 172.30.8.74
+    opnfv_maas_node03_power_type: ipmi
+    opnfv_maas_node03_power_user: admin
+    opnfv_maas_node03_power_password: octopus
+    opnfv_maas_node03_interface_mac: "00:25:b5:a0:00:4a"
+
+    opnfv_maas_node04_architecture: amd64/generic
+    opnfv_maas_node04_power_address: 172.30.8.73
+    opnfv_maas_node04_power_type: ipmi
+    opnfv_maas_node04_power_user: admin
+    opnfv_maas_node04_power_password: octopus
+    opnfv_maas_node04_interface_mac: "00:25:b5:a0:00:5a"
+
+    opnfv_maas_node05_architecture: amd64/generic
+    opnfv_maas_node05_power_address: 172.30.8.72
+    opnfv_maas_node05_power_type: ipmi
+    opnfv_maas_node05_power_user: admin
+    opnfv_maas_node05_power_password: octopus
+    opnfv_maas_node05_interface_mac: "00:25:b5:a0:00:6a"
diff --git a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
index 121f706..b07ae13 100644
--- a/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
+++ b/mcp/reclass/classes/cluster/all-mcp-ocata-common/opnfv/runtime.yml.template
@@ -2,8 +2,8 @@ parameters:
   _param:
     reclass_config_master: ${SALT_MASTER}

-    opnfv_fuel_maas_mcp_address: ${MAAS_IP}
-    opnfv_fuel_maas_pxe_network: ${MAAS_PXE_NETWORK}
-    opnfv_fuel_maas_pxe_address: ${MAAS_PXE_NETWORK_ROOTSTR}.3
-    opnfv_fuel_maas_pxe_iprange_start: ${MAAS_PXE_NETWORK_ROOTSTR}.5
-    opnfv_fuel_maas_pxe_iprange_end: ${MAAS_PXE_NETWORK_ROOTSTR}.250
+    opnfv_maas_mcp_address: ${MAAS_IP}
+    opnfv_maas_pxe_network: ${MAAS_PXE_NETWORK}
+    opnfv_maas_pxe_address: ${MAAS_PXE_NETWORK_ROOTSTR}.3
+    opnfv_maas_pxe_iprange_start: ${MAAS_PXE_NETWORK_ROOTSTR}.5
+    opnfv_maas_pxe_iprange_end: ${MAAS_PXE_NETWORK_ROOTSTR}.250
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
index 6bf0fbe..344ce8a 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/init.yml
@@ -17,19 +17,19 @@ parameters:
     infra_maas_database_password: opnfv_secret

     # infra service addresses
-    infra_config_address: 10.167.4.100
-    infra_config_deploy_address: 10.20.0.2
-    infra_maas_node01_address: 10.167.4.3
+    infra_config_address: ${_param:opnfv_infra_config_address}
+    infra_config_deploy_address: ${_param:reclass_config_master}
+    infra_maas_node01_address: ${_param:opnfv_infra_maas_node01_address}
     infra_maas_node01_deploy_address: ${_param:opnfv_fuel_maas_pxe_address}
-    infra_maas_node01_external_address: 10.16.0.3
-    infra_compute_node01_address: 10.167.4.141
-    infra_compute_node02_address: 10.167.4.142
-    infra_compute_node03_address: 10.167.4.143
+    infra_maas_node01_external_address: ${_param:opnfv_infra_maas_node01_external_address}
+    infra_compute_node01_address: ${_param:opnfv_infra_compute_node01_address}
+    infra_compute_node02_address: ${_param:opnfv_infra_compute_node02_address}
+    infra_compute_node03_address: ${_param:opnfv_infra_compute_node03_address}

-    infra_kvm_address: 10.167.4.140
-    infra_kvm_node01_address: 10.167.4.141
-    infra_kvm_node02_address: 10.167.4.142
-    infra_kvm_node03_address: 10.167.4.143
+    infra_kvm_address: ${_param:opnfv_infra_kvm_address}
+    infra_kvm_node01_address: ${_param:opnfv_infra_kvm_node01_address}
+    infra_kvm_node02_address: ${_param:opnfv_infra_kvm_node02_address}
+    infra_kvm_node03_address: ${_param:opnfv_infra_kvm_node03_address}

     infra_maas_node01_hostname: mas01
     infra_kvm_node01_hostname: kvm01
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
index 7550c1a..5064cff 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/kvm.yml
@@ -143,17 +143,16 @@ parameters:
           enabled: true
           proto: dhcp
           type: bridge
-          name_servers:
-          - 8.8.8.8
-          - 8.8.4.4
+          name_servers: ${_param:opnfv_name_servers}
           use_interfaces:
           - ${_param:deploy_nic}
           noifupdown: true
-        vlan300:
+        mgmt-vlan:
+          name: vlan${_param:opnfv_net_mgmt_vlan}
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:deploy_nic}.300
+          name: ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
           - ${_param:deploy_nic}
         br-ctl:
@@ -163,4 +162,4 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-          - ${_param:deploy_nic}.300
+          - ${_param:deploy_nic}.${_param:opnfv_net_mgmt_vlan}
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
index f77abea..cb7709c 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/infra/maas.yml
@@ -15,7 +15,7 @@ parameters:
     maas_admin_username: opnfv
     maas_admin_password: opnfv_secret
     maas_db_password: opnfv_secret
-    dns_server01: 8.8.4.4
+    dns_server01: ${_param:opnfv_name_servers}
     single_address: ${_param:infra_maas_node01_deploy_address}
   maas:
     region:
@@ -32,13 +32,13 @@ parameters:
         network_discovery: 'enabled'
         default_min_hwe_kernel: 'hwe-16.04'
       subnets:
-        opnfv_fuel_maas_pxe:
-          name: ${_param:opnfv_fuel_maas_pxe_network}/24
-          cidr: ${_param:opnfv_fuel_maas_pxe_network}/24
+        opnfv_maas_pxe:
+          name: ${_param:opnfv_maas_pxe_network}/24
+          cidr: ${_param:opnfv_maas_pxe_network}/24
           gateway_ip: ${_param:single_address}
           iprange:
-            start: ${_param:opnfv_fuel_maas_pxe_iprange_start}
-            end: ${_param:opnfv_fuel_maas_pxe_iprange_end}
+            start: ${_param:opnfv_maas_pxe_iprange_start}
+            end: ${_param:opnfv_maas_pxe_iprange_end}
             type: dynamic
           vlans:
             untagged:
@@ -48,57 +48,57 @@ parameters:
       machines:
         kvm01:
           interface:
-            mac: "00:25:b5:a0:00:2a"
+            mac: ${_param:opnfv_maas_node01_interface_mac}
           power_parameters:
-            power_address: "172.30.8.75"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node01_power_address}
+            power_password: ${_param:opnfv_maas_node01_power_password}
+            power_type: ${_param:opnfv_maas_node01_power_type}
+            power_user: ${_param:opnfv_maas_node01_power_user}
+          architecture: ${_param:opnfv_maas_node01_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         kvm02:
           interface:
-            mac: "00:25:b5:a0:00:3a"
+            mac: ${_param:opnfv_maas_node02_interface_mac}
           power_parameters:
-            power_address: "172.30.8.65"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node02_power_address}
+            power_password: ${_param:opnfv_maas_node02_power_password}
+            power_type: ${_param:opnfv_maas_node02_power_type}
+            power_user: ${_param:opnfv_maas_node02_power_user}
+          architecture: ${_param:opnfv_maas_node02_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         kvm03:
           interface:
-            mac: "00:25:b5:a0:00:4a"
+            mac: ${_param:opnfv_maas_node03_interface_mac}
           power_parameters:
-            power_address: "172.30.8.74"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node03_power_address}
+            power_password: ${_param:opnfv_maas_node03_power_password}
+            power_type: ${_param:opnfv_maas_node03_power_type}
+            power_user: ${_param:opnfv_maas_node03_power_user}
+          architecture: ${_param:opnfv_maas_node03_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         cmp001:
           interface:
-            mac: "00:25:b5:a0:00:5a"
+            mac: ${_param:opnfv_maas_node04_interface_mac}
           power_parameters:
-            power_address: "172.30.8.73"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node04_power_address}
+            power_password: ${_param:opnfv_maas_node04_power_password}
+            power_type: ${_param:opnfv_maas_node04_power_type}
+            power_user: ${_param:opnfv_maas_node04_power_user}
+          architecture: ${_param:opnfv_maas_node04_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
         cmp002:
           interface:
-            mac: "00:25:b5:a0:00:6a"
+            mac: ${_param:opnfv_maas_node05_interface_mac}
           power_parameters:
-            power_address: "172.30.8.72"
-            power_password: "octopus"
-            power_type: ipmi
-            power_user: "admin"
-          architecture: 'amd64/generic'
+            power_address: ${_param:opnfv_maas_node05_power_address}
+            power_password: ${_param:opnfv_maas_node05_power_password}
+            power_type: ${_param:opnfv_maas_node05_power_type}
+            power_user: ${_param:opnfv_maas_node05_power_user}
+          architecture: ${_param:opnfv_maas_node05_architecture}
           distro_series: xenial
           hwe_kernel: hwe-16.04
   linux:
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
index 93a26b1..a48d924 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/compute.yml
@@ -67,19 +67,19 @@ parameters:
           netmask: 255.255.255.0
           mtu: 1500
           use_interfaces:
-          - ${_param:tenant_nic}.302
-        vlan300:
+          - ${_param:tenant_nic}.${_param:opnfv_net_tenant_vlan}
+        mgmt-vlan:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:mgmt_nic}.300
+          name: ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
           use_interfaces:
           - ${_param:mgmt_nic}
-        vlan302:
+        tenant-vlan:
           enabled: true
           proto: manual
           type: vlan
-          name: ${_param:tenant_nic}.302
+          name: ${_param:tenant_nic}.${_param:opnfv_net_tenant_vlan}
           use_interfaces:
           - ${_param:tenant_nic}
         br-ctl:
@@ -89,7 +89,7 @@ parameters:
           address: ${_param:single_address}
           netmask: 255.255.255.0
           use_interfaces:
-          - ${_param:mgmt_nic}.300
+          - ${_param:mgmt_nic}.${_param:opnfv_net_mgmt_vlan}
         br-floating:
           enabled: true
           type: ovs_bridge
diff --git a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
index 9c2bc7b..eff5651 100644
--- a/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
+++ b/mcp/reclass/classes/cluster/baremetal-mcp-ocata-ovs-ha/openstack/init.yml
@@ -3,32 +3,32 @@ parameters:

     openstack_version: ocata

-    openstack_gateway_node01_address: 10.167.4.124
-    openstack_gateway_node02_address: 10.167.4.125
-    openstack_gateway_node03_address: 10.167.4.126
-    openstack_gateway_node01_tenant_address: 10.1.0.6
-    openstack_gateway_node02_tenant_address: 10.1.0.7
-    openstack_gateway_node03_tenant_address: 10.1.0.9
+    openstack_gateway_node01_address: ${_param:opnfv_openstack_gateway_node01_address}
+    openstack_gateway_node02_address: ${_param:opnfv_openstack_gateway_node02_address}
+    openstack_gateway_node03_address: ${_param:opnfv_openstack_gateway_node03_address}
+    openstack_gateway_node01_tenant_address: ${_param:opnfv_openstack_gateway_node01_tenant_address}
+    openstack_gateway_node02_tenant_address: ${_param:opnfv_openstack_gateway_node02_tenant_address}
+    openstack_gateway_node03_tenant_address: ${_param:opnfv_openstack_gateway_node03_tenant_address}
     openstack_gateway_node01_hostname: gtw01
     openstack_gateway_node02_hostname: gtw02
     openstack_gateway_node03_hostname: gtw03

     # openstack service addresses
-    openstack_proxy_address: 10.167.4.80
-    openstack_proxy_node01_address: 10.167.4.81
-    openstack_proxy_node02_address: 10.167.4.82
-    openstack_control_address: 10.167.4.10
-    openstack_control_node01_address: 10.167.4.11
-    openstack_control_node02_address: 10.167.4.12
-    openstack_control_node03_address: 10.167.4.13
-    openstack_database_address: 10.167.4.50
-    openstack_database_node01_address: 10.167.4.51
-    openstack_database_node02_address: 10.167.4.52
-    openstack_database_node03_address: 10.167.4.53
-    openstack_message_queue_address: 10.167.4.40
-    openstack_message_queue_node01_address: 10.167.4.41
-    openstack_message_queue_node02_address: 10.167.4.42
-    openstack_message_queue_node03_address: 10.167.4.43
+    openstack_proxy_address: ${_param:opnfv_openstack_proxy_address}
+    openstack_proxy_node01_address: ${_param:opnfv_openstack_proxy_node01_address}
+    openstack_proxy_node02_address: ${_param:opnfv_openstack_proxy_node02_address}
+    openstack_control_address: ${_param:opnfv_openstack_control_address}
+    openstack_control_node01_address: ${_param:opnfv_openstack_control_node01_address}
+    openstack_control_node02_address: ${_param:opnfv_openstack_control_node02_address}
+    openstack_control_node03_address: ${_param:opnfv_openstack_control_node03_address}
+    openstack_database_address: ${_param:opnfv_openstack_database_address}
+    openstack_database_node01_address: ${_param:opnfv_openstack_database_node01_address}
+    openstack_database_node02_address: ${_param:opnfv_openstack_database_node02_address}
+    openstack_database_node03_address: ${_param:opnfv_openstack_database_node03_address}
+    openstack_message_queue_address: ${_param:opnfv_openstack_message_queue_address}
+    openstack_message_queue_node01_address: ${_param:opnfv_openstack_message_queue_node01_address}
+    openstack_message_queue_node02_address: ${_param:opnfv_openstack_message_queue_node02_address}
+    openstack_message_queue_node03_address: ${_param:opnfv_openstack_message_queue_node03_address}


     openstack_telemetry_hostname: mdb
@@ -36,23 +36,23 @@ parameters:
     openstack_telemetry_node02_hostname: mdb02
     openstack_telemetry_node03_hostname: mdb03

-    openstack_telemetry_address: 10.167.4.75
-    openstack_telemetry_node01_address: 10.167.4.76
-    openstack_telemetry_node02_address: 10.167.4.77
-    openstack_telemetry_node03_address: 10.167.4.78
+    openstack_telemetry_address: ${_param:opnfv_openstack_telemetry_address}
+    openstack_telemetry_node01_address: ${_param:opnfv_openstack_telemetry_node01_address}
+    openstack_telemetry_node02_address: ${_param:opnfv_openstack_telemetry_node02_address}
+    openstack_telemetry_node03_address: ${_param:opnfv_openstack_telemetry_node03_address}

     # OpenStack Compute
-    openstack_compute_node01_single_address: 10.167.4.101
-    openstack_compute_node02_single_address: 10.167.4.102
-    openstack_compute_node03_single_address: 10.167.4.103
-    openstack_compute_node01_control_address: 10.167.4.101
-    openstack_compute_node02_control_address: 10.167.4.102
-    openstack_compute_node03_control_address: 10.167.4.103
-    openstack_compute_node01_tenant_address: 10.1.0.101
-    openstack_compute_node02_tenant_address: 10.1.0.102
-    openstack_compute_node03_tenant_address: 10.1.0.103
-    openstack_compute_node01_external_address: 172.30.10.2
-    openstack_compute_node02_external_address: 172.30.10.3
+    openstack_compute_node01_single_address: ${_param:opnfv_openstack_compute_node01_single_address}
+    openstack_compute_node02_single_address: ${_param:opnfv_openstack_compute_node02_single_address}
+    openstack_compute_node03_single_address: ${_param:opnfv_openstack_compute_node03_single_address}
+    openstack_compute_node01_control_address: ${_param:opnfv_openstack_compute_node01_control_address}
+    openstack_compute_node02_control_address: ${_param:opnfv_openstack_compute_node02_control_address}
+    openstack_compute_node03_control_address: ${_param:opnfv_openstack_compute_node03_control_address}
+    openstack_compute_node01_tenant_address: ${_param:opnfv_openstack_compute_node01_tenant_address}
+    openstack_compute_node02_tenant_address: ${_param:opnfv_openstack_compute_node02_tenant_address}
+    openstack_compute_node03_tenant_address: ${_param:opnfv_openstack_compute_node03_tenant_address}
+    openstack_compute_node01_external_address: ${_param:opnfv_openstack_compute_node01_external_address}
+    openstack_compute_node02_external_address: ${_param:opnfv_openstack_compute_node02_external_address}

     # openstack service hostnames
     openstack_proxy_hostname: prx
